packages/core-components/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+import Paper, { PaperProps } from '@material-ui/core/Paper';
+import Popper, { PopperProps } from '@material-ui/core/Popper';
+import TextField, { OutlinedTextFieldProps } from '@material-ui/core/TextField';
+import Grow from '@material-ui/core/Grow';
+import {
+  createStyles,
+  makeStyles,
+  Theme,
+  withStyles,
+} from '@material-ui/core/styles';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Autocomplete, { AutocompleteProps } from '@material-ui/lab/Autocomplete';
+import React, { ReactNode } from 'react';
+
+const useStyles = makeStyles(
+  theme => ({
+    root: {},
+    label: {
+      position: 'relative',
+      fontWeight: 'bold',
+      fontSize: theme.typography.body2.fontSize,
+      fontFamily: theme.typography.fontFamily,
+      color: theme.palette.text.primary,
+      '& > span': {
+        top: 0,
+        left: 0,
+        position: 'absolute',
+      },
+    },
+    input: {},
+  }),
+  { name: 'BackstageAutocomplete' },
+);
+
+const BootstrapAutocomplete = withStyles(
+  (theme: Theme) =>
+    createStyles({
+      root: {},
+      paper: {
+        margin: 0,
+      },
+      hasClearIcon: {},
+      hasPopupIcon: {},
+      focused: {},
+      inputRoot: {
```

Comment:
Yeh unfortunately there's a lot of tweaking here to get the component to look like pixel perfect like the `Select` component. I think this is because the `Select` component has a bit of customization too.